go-controller/pkg/ovn/controller/services/node_tracker.go

```diff
@@ -0,0 +1,211 @@
+package services
+
+import (
+	"net"
+	"reflect"
+	"sort"
+	"sync"
+
+	globalconfig "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	v1 "k8s.io/api/core/v1"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+// nodeTracker watches all Node objects and maintains a cache of information relevant
+// to service creation. If a new node is created, it requests a resync of all services,
+// since need to apply those service's load balancers to the new node as well.
+type nodeTracker struct {
+	sync.Mutex
+
+	// nodes is the list of nodes we know about
+	// map of name -> info
+	nodes map[string]nodeInfo
```

Comment:
I suspect that will hurt performance; the vast majority of accesses will be via `nodeTracker.allNodes()`, which deep-copies the map. sync.Map is only a performance gain when the concurrent accesses are non-colliding.

It may be a performance improvement to switch to RWMutex. Either way, I do not expect this mutex to be particularly hot; the critical path has no calls to ovsdb.

Reply:
RWMutex has bad performance, I wouldn't use that. sync map is easier here because then you dont have to keep track of the mutex. It's a minor thing either way.